enhancements/sig-architecture/105-aws-iam-registration/README.md

```diff
@@ -0,0 +1,464 @@
+
+## Summary
+
+OCM's use of a CSR based mechanism for registering managed clusters with the hub cluster is incompatible with Kubernetes environments that cannot issue client auth certificates such as Amazon Elastic Kubernetes Service (EKS).
+This enhancement provides an AWS IAM based registration mechanism so that OCM can support EKS-based hub clusters.
+
+## Motivation
+
+For cluster administrators, it is often preferable to leverage their cloud provider’s managed Kubernetes service (e.g. AKS, EKS, IKS, GKE etc) rather than self-managing the Kubernetes control plane and worker nodes - reducing cluster management overhead and complexity.
+However, in some of these environments (for example EKS), the use of CSRs for client authentication is unsupported.
+
+Adopting OCM should not require its users to change how they deploy and manage their hub Kubernetes cluster.
+
+One option is to run [multicluster-controlplane](https://github.com/open-cluster-management-io/multicluster-controlplane) service inside or outside Kubernetes and use that as a hub. However, multicluster-controlplane comes with following limitations:
+- It is not a full Kubernetes controlplane and hence some OCM addons might not work
+- The user is responsible for maintaining HA etcd cluster, instead of being able to leverage managed etcd that comes with EKS controlplane. 
+
+As such, OCM needs to support an alternative registration mechanism to CSRs that is expected to work in EKS, providing maximum compatibility with managed EKS.
+
+### Goals
+
+- The OCM hub cluster can run on EKS.
+- `clusteradm` tooling provides options to select aws-irsa registration while initializing the hub as well as joining a managed cluster to the hub.
+- Make authentication strategy pluggable, making it easier for other cloud providers to add support for their platform in the future.
+
+### Non-Goals
+
+- Support for running a hub cluster on other cloud providers like Google Cloud Platform GKE and Azure AKS, even though this enhancement will make future support for other cloud providers easier.
+- Support for EKS cluster as hub and any other cloud provider managed kubernetes service as a managed cluster. This enhancement aims at using EKS for both managed as well as hub cluster.
+
+## Proposal
+
+### User Stories
+
+#### Story 1 - Hub administrator initializes a hub cluster using AWS IRSA authentication strategy
+
+It must be possible for the cluster administrator to specify they wish to authenticate registration requests using `aws-irsa` authentication strategy in the `clusteradm init` command. The default authentication strategy will be `csr`:
+
+```
+% clusteradm init \
+     --wait \
+     --registration-auth=aws-irsa \
+     --context ${CTX_HUB_CLUSTER}
+```
+
+
+#### Story 2 - EKS Hub administrator must initialize a hub using aws-irsa authentication strategy only, using csr authentication strategy will throw an error
```

Comment:
I agree it's reasonable to specify a list of allowed authentication strategies, but could it be a positive list that allows

1. aws-irsa only
2. csr only
3. aws-irsa and/or csr

Reply:
Agreed with #1 and 2, but not sure about the third option because our solution is dependent on IRSA which is an EKS feature and both hub and spoke will need to assume AWS IAM role through IRSA. csr solution wont work on EKS and aws-irsa solution won't work outside EKS, so i believe they are exclusive solutions and mix and match might not work.

However, please share your thoughts and let us know if you can think of any scenario where both solutions can work together.

Reply:
Can you help describe the IRSA capability that is missing in non-EKS capabilities?  Is the interpretation of AWS tokens somehow not possible?  Is the user mapping not done?  I would expect the ability to interpret AWS tokens to be generally possible so different clusters could inter-operate.

Reply:
Updated the proposal to list all the options under story 2

Reply:
maybe some rewording here. It is not necessary to aws only and throw error for csr in this case. The user certainly could select to only enable aws-isra

Reply:
Thanks @qiujian16 ,
I just updated the PR. Please let me know if there is anything missing.